### ISSUE TITLE
Plugin stopped working this morning

### ISSUE THREAD

**Issue by rjmcfadd** [KraigM/homebridge-nest#167]
This morning the Nest plugin caused homebridge to stop working. I tried to re-add my thermostat, getting a new token but that didn't work. If I remove nest from my config file, homebridge starts working again.

**Comment by Krocko** [comment]
Same for me. See #125

**Comment by Judas2016** [comment]
Same for me, Homebridge is hanging, as a result, won't fully start. Noticed the API version in my dashboard now shows v6, something changed.  I stripped out the code, token, clientid, clientsecret expecting the plugin to default to legacy API, that does not appear to happen now, it thows the following error, does not appear to fall back. 

 Error: You did not specify {'token'} or {'clientId','clientSecret','code'}, one set of which is required for the new API

**Comment by cgilis** [comment]
Same problem here, tried to reinstall entire RPi did not work :(

**Comment by chrisjshull** [comment]
It looks like the legacy API was quietly removed front the version pushed to NPM. :/

**Comment by ghost** [comment]
@chrisjshull Would you take a stab at setting me straight on new vs. old API? When I added my thermostat to my homebridge instance (probably 2-3 weeks ago), I followed the instructions for "How to Setup New API" on this repos home page. Also, over at #125 , it seems like there's a workaround to fallback to what @PencilDH is referring to as the Legacy API. Obviously something has changed and it sounds like people are starting to notice, I'm just trying to get a grasp of the terminology being thrown around in relation to what I know I have done/tried with my own setup. Thanks!

**Comment by ghost** [comment]
Any updates about this?

**Comment by chrisjshull** [comment]
OK, a couple of things I'd try:

1. use my fork, it has a lot of fixes from people merged in: `npm install -g "https://github.com/chrisjshull/homebridge-nest"`
1. run homebridge with debug logging: `DEBUG=* homebridge -D`
1. I don't think this is the issue people are seeing, but it should be noted that the NEST API has a rate limit that will just reject commands for a while if you trigger it.

Let me know how that goes, and share whatever logging you get.

**Comment by chrisjshull** [comment]
hmm. @cgilis, everything seems alright in that log. Also seems to have got past the point in your earlier screenshot. (Are you sure it's still stuck? How can you tell?)

Sanity check - can you you share:
`homebridge --version`
`node --version`
`npm --version`

I'm also not seeing a log I think my fork added. Any you sure you installed it. Maybe it's not being used because of:
> Warning: skipping plugin found at '/home/pi/.npm-global/lib/node_modules/homebridge-nest' since we already loaded the same plugin from '/usr/lib/node_modules/homebridge-nest'.

**Comment by cgilis** [comment]
@chrisjshull because I don's see it appearing in my Home app.. maybe I need to refresh something?

![screen shot 2018-02-09 at 10 24 42](https://user-images.githubusercontent.com/5811658/36020591-89f6cdb4-0d83-11e8-9b3b-df5ae54801bf.png)

**Comment by ghost** [comment]
It works again, i was to fast reinstall the plugin on developers.nest.com (forgot to activate the thermostat itself) thx for the help

**Comment by ghost** [comment]
@chrisjshull I used your fork and it resolved my issue with missing thermostats. As I have 3 Nests, I end up with a bunch of accessories listed (3 Eco switches, 3 Fans, 3 Home/Away switches, and 3 Thermostats). Is this normal, or just a limitation of Homekit?

One bug I noticed, asking Siri the current temp, she accurately reports the temp of the requested thermostat, but also reports 'going down to....' and references the Eco low temp setting. So somehow the Eco settings are being mixed up with the current high/low settings.
Example: My Eco setting is 50 Low 80 High. With a Thermostat home setting of 66 low and 75 high on Auto, Siri reports current temp 68 going down to 50...

**Comment by chrisjshull** [comment]
> @chrisjshull I used your fork and it resolved my issue with missing thermostats. As I have 3 Nests, I end up with a bunch of accessories listed (3 Eco switches, 3 Fans, 3 Home/Away switches, and 3 Thermostats). Is this normal, or just a limitation of Homekit?

@Chivs, yes, in order to expose the fan, eco, etc settings in the built-in Home app some items needed to be exposed as if they were separate devices. Personally, I keep only the bits I care about in my favorites, but a setting could be exposed if you really wanted them gone (let me know at https://github.com/chrisjshull/homebridge-nest/issues/new). 

> One bug I noticed, asking Siri the current temp, she accurately reports the temp of the requested thermostat, but also reports 'going down to....' and references the Eco low temp setting. So somehow the Eco settings are being mixed up with the current high/low settings.
Example: My Eco setting is 50 Low 80 High. With a Thermostat home setting of 66 low and 75 high on Auto, Siri reports current temp 68 going down to 50...

Great find. Can you also file a separate ticket for this at https://github.com/chrisjshull/homebridge-nest/issues/new?
